files/default/neutron-ha-tool.py

```diff
@@ -237,9 +237,9 @@ def run(args):
     qclient.format = 'json'
 
     if args.agent_selection_mode == 'random':
-        agent_picker_class = RandomAgentPicker
+        agent_picker = RandomAgentPicker()
```

Comment:
Reviewing the commit message:

> We create the AgentPicker instances at setup time, only filling in the
> agent list before the picking begins.

Fine so far.

> This way clients are not
> responsible for creating the AgentPicker instances, thus the
> dependencies are hidden from clients.

I think you should avoid using the word "clients" here, because that sounds like it refers to neutron client instances, and presumably you don't mean that.  Actually I don't know what you mean by "clients" here :-/  Presumably not "callers", because you are making the callers responsible for creating the instances, and that's the right thing to do.  Maybe I'm being stupid but I don't get it.

IOW I agree with the code changes, just can't understand this last sentence.